packages/core/src/node/backend-application.ts

```diff
@@ -102,13 +114,49 @@ export class BackendApplication {
         this.app.use(...handlers);
     }
 
-    async start(aPort?: number, aHostname?: string): Promise<http.Server> {
-        const deferred = new Deferred<http.Server>();
-        let server: http.Server;
+    async start(aPort?: number, aHostname?: string): Promise<http.Server | https.Server> {
+        const deferred = new Deferred<http.Server | https.Server>();
+        let server: http.Server | https.Server;
         const port = aPort !== undefined ? aPort : this.cliParams.port;
         const hostname = aHostname !== undefined ? aHostname : this.cliParams.hostname;
-        server = this.app.listen(port, hostname!, () => {
-            this.logger.info(`Theia app listening on http://${hostname || 'localhost'}:${server.address().port}.`);
+
+        if (this.cliParams.ssl) {
+
+            if (this.cliParams.cert === undefined) {
+                throw new Error("Missing --cert option, see --help for usage");
+            }
+
+            if (this.cliParams.certkey === undefined) {
+                throw new Error("Missing --certkey option, see --help for usage");
+            }
+
+            let key;
+            let cert;
+            try {
+                key = await fs.readFile(this.cliParams.certkey as string);
+            } catch (err) {
+                await this.logger.error(`Can't read certificate key`);
```

Comment:
It would be good to log the reason the file couldn't be read (I assume it's some field in err).

Reply:
Yes I trow the error after that and it's logged upstream by this code:

```
module.exports = (port, host) => Promise.resolve()${this.compileBackendModuleImports(backendModules)}
    .then(() => start(port, host)).catch(reason => {
        console.error('Failed to start the backend application.');
        if (reason) {
            console.error(reason);
        }
        throw reason;
    });`;
```

in server.js

Reply:
You are using template string without any variable substitution.  Is there an overhead to using this vs a regular string, when there isn't any variable substitution in the string?

    `Hello`

vs

    'Hello'

?

Reply:
@simark sorry that's because I took the sting out of the generator code...

The key difference is that even without substitution this supports multi-line strings.

Reply:
But I don't know about performance I assume there is a small overhead